### PR TITLE
Add chaos testing guide content

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
- Adds Chaos Testing Guide to help users build resilient,
reliable and performant services.
- Supports ADR: https://github.com/operate-first/sre/pull/12.

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
@david-martin PTAL. Thanks.

**Comment by david-martin** [comment]
@chaitanyaenr The content looks good.
I'm less sure if it renders OK. There seems to be netlify integration but maybe that needs the CI job to pass first?
I had a look at the CI failure but it's not immediately obvious what's wrong.

@quaid or @durandom Can you guide us here on how to get a netlify preview of the blog post?

**Comment by david-martin** [comment]
@chaitanyaenr The content looks good.
I'm less sure if it renders OK. There seems to be netlify integration but maybe that needs the CI job to pass first?
I had a look at the CI failure but it's not immediately obvious what's wrong.

@quaid or @durandom Can you guide us here on how to get a netlify preview of the blog post?

**Comment by durandom** [comment]
I'm just fixing the netlify bits in https://github.com/operate-first/blog/pull/3
maybe we need to close/open this PR

**Comment by durandom** [comment]
@chaitanyaenr @david-martin https://github.com/operate-first/blog/pull/7 has netlify. maybe force pushing to this PR works, or you open another PR. Sorry

**Comment by quaid** [comment]
/assign @quaid

I'm doing a review based on the in-draft process, template, and style guides (content and technical markup).

[Blog publishing process and blog post template](https://github.com/operate-first/blog/pull/5/files)
[The technical markup style guide](https://github.com/operate-first/community-handbook/pull/43/files#diff-03a962830a7caf3e81eb78a229039097f58e5efa96a16cc8fa783ed332f43afd)
[Writing style guide](https://github.com/operate-first/community-handbook/blob/main/contributing/style-guide.md)

File: content/posts/chaos-testing-guide/index.mdx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    - Establish and define your steady state and metrics - understand the behavior and performance under stable conditions  and define the metrics that will be used to evaluate the system’s behavior.  Then decide on acceptable outcomes before injecting chaos. 
+    - Analyze the statuses and metrics of all components during the chaos test runs.
+    - Improve the areas that are not resilient and performant by comparing the key metrics and Service Level Objectives (SLOs) to the stable conditions before the chaos.
+      For example: evaluating the API server latency or application uptime to see if the key performance indicators and service level indicators are still within the acceptable limits.
+
+
+### Best Practices
+Now that we understand the test methodology, let’s take a look at the best practices for an OpenShift cluster.  On that platform there are user applications and cluster workloads that need to be designed for stability and to provide the best user experience possible:
+- Alerts with appropriate severity should get fired
```
Comment (quaid) [inline content/posts/chaos-testing-guide/index.mdx:72]: The sentence construct is a bit confusing in two ways:  if you misinterpret 'fired' as "end the employment of an alert", and if read as, "Fire the alerts that have appropriate severity," I'm not sure it makes sense even in an SRE context?

That is, if it's already an alert, hasn't an alert been fired? What does it mean to "fire an alert" when the alert exists already?

Perhaps "alert" is doing double-duty in this sentence? If so, does this or something similar work?
```suggestion
- System alerts with appropriate severity should trigger a human response.
```

File: content/posts/chaos-testing-guide/index.mdx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  - We want to look at this in terms of CPU, Memory, Storage, Throughput, Network etc.
+
+- Minimal initialization time and fast recovery logic
+  - The controller watching the component should recognize a failure as soon as possible. The component needs to have minimal initialization time to avoid extended downtime or overloading the replicas if it is a highly available configuration. The cause of failure can be because of issues with the infrastructure on top of which it’s running, application failures or because of service failures that it depends on.
+
+- Appropriate CPU/Memory limits set to avoid performance throttling and OOM kills
+  - There might be rogue applications hogging resources ( CPU/Memory ) on the nodes which might lead to applications underperforming  or worse getting OOM killed. It’s important to ensure that the applications and system components have reserved resources for the kube-scheduler to take into consideration in order to keep them performing at the expected levels.
```
Comment (quaid) [inline content/posts/chaos-testing-guide/index.mdx:84]:
```suggestion
  - There might be rogue applications hogging resources ( CPU/Memory ) on the nodes which might lead to applications underperforming  or worse getting OOM killed. It’s important to ensure that the applications and system components have reserved resources for the `kube-scheduler` to take into consideration in order to keep them performing at the expected levels.
```

The (proposed) markup style guide uses `backticks` for application or program executable names.

File: content/posts/chaos-testing-guide/index.mdx (hunk 1 above)
Comment (chaitanyaenr) [reply content/posts/chaos-testing-guide/index.mdx:72]: Makes sense, tweaked it to "Alerts with appropriate severity". Thanks.

File: content/posts/chaos-testing-guide/index.mdx (hunk 2 above)
Comment (chaitanyaenr) [reply content/posts/chaos-testing-guide/index.mdx:84]: Fixed it, thanks.

**Comment by chaitanyaenr** [comment]
@quaid, I updated the commit to bake in your suggestions. Also, fixed the errors reported by pre-commit. PTAL. Thanks.

**Comment by quaid** [comment]
Thanks @chaitanyaenr, I lost track of time this week but let's see if we can resolve this.

/retest

**Comment by quaid** [comment]
@durandom this is breaking in the toolchain and I'm not sure if it's syntax in the post breaking the build or the toolchain itself, can you investigate?

**Comment by durandom** [comment]
@chaitanyaenr I created https://github.com/chaitanyaenr/blog/pull/1
I rebased on top of the current main branch and fixed the `frontmatter` section. 
You also just might want to cherry-pick https://github.com/chaitanyaenr/blog/pull/1/commits/0d947e096c89cdd3e756eb582d97af3b8b950564

**Pull request merged**